lib/asn1/template.c

```diff
@@ -3015,7 +3015,7 @@ _asn1_copy(const struct asn1_template *t, const void *from, void *to)
 	    unsigned int i;
 
 	    tel->val = calloc(fel->len, ellen);
-	    if (tel->val == NULL)
+	    if (tel->val == NULL && fel->len * ellen > 0)
```

Comment:
Let's avoid the potential for overflowing here and not multiply.  `ellen` will never be zero, so all we have to check for is that `fel->len` is not zero.

Reply:
thanks, applied